CodeConverter/CSharp/ExpressionNodeVisitor.cs

```diff
@@ -97,17 +97,45 @@ public override async Task<CSharpSyntaxNode> DefaultVisit(SyntaxNode node)
-            return InterpolatedString(interpolationsList);
+            _extraUsingDirectives.Add("System.Xml.Linq");
+            var arguments = SyntaxFactory.SeparatedList(
+                (await node.PrecedingMisc.SelectAsync(async misc => SyntaxFactory.Argument(await misc.AcceptAsync<ExpressionSyntax>(TriviaConvertingExpressionVisitor))))
+                .Concat(SyntaxFactory.Argument(await node.Root.AcceptAsync<ExpressionSyntax>(TriviaConvertingExpressionVisitor)).Yield())
+                .Concat(await node.FollowingMisc.SelectAsync(async misc => SyntaxFactory.Argument(await misc.AcceptAsync<ExpressionSyntax>(TriviaConvertingExpressionVisitor))))
+                );
+            return SyntaxFactory.ObjectCreationExpression(SyntaxFactory.IdentifierName("XDocument")).WithArgumentList(SyntaxFactory.ArgumentList(arguments));
         }
 
         public override async Task<CSharpSyntaxNode> VisitXmlElement(VBasic.Syntax.XmlElementSyntax node)
         {
             _extraUsingDirectives.Add("System.Xml.Linq");
-            var interpolationsList = SyntaxFactory.List(await AcceptXmlInterpolatedAsync(node));
-            return InterpolatedString(interpolationsList);
+            var arguments = SyntaxFactory.SeparatedList(
+                SyntaxFactory.Argument(CommonConversions.Literal(node.StartTag.Name.ToString())).Yield()
+                .Concat(await node.StartTag.Attributes.SelectAsync(async attribute => SyntaxFactory.Argument(await attribute.AcceptAsync<ExpressionSyntax>(TriviaConvertingExpressionVisitor))))
+                .Concat(await node.Content.SelectAsync(async content => SyntaxFactory.Argument(await content.AcceptAsync<ExpressionSyntax>(TriviaConvertingExpressionVisitor))))
+                );
+            return SyntaxFactory.ObjectCreationExpression(SyntaxFactory.IdentifierName("XElement")).WithArgumentList(SyntaxFactory.ArgumentList(arguments));
+        }
+
+        public async override Task<CSharpSyntaxNode> VisitXmlEmptyElement(VBSyntax.XmlEmptyElementSyntax node)
+        {
+            _extraUsingDirectives.Add("System.Xml.Linq");
+            var arguments = SyntaxFactory.SeparatedList(
+                SyntaxFactory.Argument(CommonConversions.Literal(node.Name.ToString())).Yield()
+                .Concat(await node.Attributes.SelectAsync(async attribute => SyntaxFactory.Argument(await attribute.AcceptAsync<ExpressionSyntax>(TriviaConvertingExpressionVisitor))))
+                );
+            return SyntaxFactory.ObjectCreationExpression(SyntaxFactory.IdentifierName("XElement")).WithArgumentList(SyntaxFactory.ArgumentList(arguments));
         }
 
+        public override async Task<CSharpSyntaxNode> VisitXmlAttribute(VBasic.Syntax.XmlAttributeSyntax node)
+        {
+            var arguments = SyntaxFactory.SeparatedList(
+                SyntaxFactory.Argument(CommonConversions.Literal(node.Name.ToString())).Yield()
```

Comment:
It's usually preferable to specify which exact string you want - in this library calling ToString can have surprising effects for various node types or for various escaped values. As far as I know, all xml tag names are valid in a C# literal with no extra escaping though so I think this will probably be fine.